### ISSUE TITLE
Invalid treatment for executes with columns

### ISSUE THREAD

**Issue by Folling** [rusqlite/rusqlite#645]
Rusqlite currently returns an error when `execute` returns no row, but has columns.
However, both renaming a table, as well as renaming a column (which ought to be fine usecases for `execute`) return no rows, but have a column in the result set.

I'm uncertain why they do, but it's something I've noticed. It might have been a regression with the recent changes to how renames are propagated throughout the database.

This is with version 0.21.0 of rusqlite, and version 3.31.1 of sqlite.

**Comment by Folling** [comment]
Would you mind explaining why you believe this issue to be invalid?
There is an auxiliary check from within `execute_with_bound_parameters`, which is performed when there is no row in the result set, which checks whether columns are there or not, seemingly unnecessary. If you agree with that statement, that's fine - but please at least elaborate on it.

Ref:
```rs
    fn execute_with_bound_parameters(&mut self) -> Result<usize> {
        self.check_update()?;
        let r = self.stmt.step();
        self.stmt.reset();
        match r {
            ffi::SQLITE_DONE => {
                if self.column_count() == 0 { // <--- why is this check necessary?
                    Ok(self.conn.changes())
                } else {
                    Err(Error::ExecuteReturnedResults)
                }
            }
            ffi::SQLITE_ROW => Err(Error::ExecuteReturnedResults),
            _ => Err(self.conn.decode_result(r).unwrap_err()),
        }
    }
```

**Comment by gwenn** [comment]
This issue was `invalid` because some basic informations were missing like a stacktrace or a minimal example to reproduce the error.
And, as far as I know, you cannot bind parameter in an `ALTER TABLE` or any DDL.
Also, the number of rows impacted is meaningless...
So [execute_batch](https://docs.rs/rusqlite/0.21.0/rusqlite/struct.Connection.html#method.execute_batch) should be used.

I will check why the column count is not zero.
Regards.

**Comment by gwenn** [comment]
Minimal test to reproduce:
```rust
        #[test]
        fn test_alter_table() {
            let db = checked_memory_handle();
            db.execute_batch("CREATE TABLE x(t);").unwrap();
            db.execute("ALTER TABLE x RENAME TO y;", params![]).unwrap();
        }
```

**Comment by gwenn** [comment]
https://bugs.php.net/bug.php?id=78813

**Comment by gwenn** [comment]
```rust
        #[test]
        fn test_alter_table() {
            let db = checked_memory_handle();
            db.execute_batch("CREATE TABLE x(t);").unwrap();
            let mut stmt = db.prepare("ALTER TABLE x RENAME TO y;").unwrap();
            let rows = stmt.query(params![]).unwrap();
            assert_eq!(Some(1), rows.column_count());
            assert_eq!(Some(Ok("1")), rows.column_name(0));
        }
```

**Comment by Folling** [comment]
Ah, I see!
Pardon me, I had forgotten to read through the issue-creation guide for this specific repository and thought that the description of the concept of the issue was sophisticated enough!

**Comment by gwenn** [comment]
Fixed by version 0.22.0
